### PR TITLE
Add support for Sql IN predicate.

### PR DESCRIPTION
**Pull request by chakankardb**
This PR adds support for Sql IN predicate.
We limit the size of the list to guard against pathological inputs.

### CODE REVIEW

File: spark/src/main/scala/io/delta/sharing/spark/filters/OpConverter.scala [hunk 1]
```diff
@@ -53,6 +53,8 @@ import org.apache.spark.sql.types.{
 // should skip filtering in such cases.
 object OpConverter {
 
+  val kMaxSqlInOpSizeLimit = 20
```
Comment (zhuansunxt) [inline spark/src/main/scala/io/delta/sharing/spark/filters/OpConverter.scala:56]: Can you add some documentation to the definition of this limit?

**Review by linzhou-db — APPROVED** [review]

File: spark/src/test/scala/io/delta/sharing/spark/filters/OpConverterSuite.scala [hunk 2]
```diff
@@ -275,4 +275,32 @@ class OpConverterSuite extends SparkFunSuite {
...
+    val op = OpConverter.convert(Seq(sqlIn)).get
+    op.validate()
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "22"))) == false)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "23"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "24"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "25"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "26"))) == true)
+    assert(op.evalExpectBoolean(EvalContext(Map("userId" -> "27"))) == false)
+
+    var sqlLiterals2: Seq[SqlExpression] = Seq.empty
```
Comment (zhuansunxt) [inline spark/src/test/scala/io/delta/sharing/spark/filters/OpConverterSuite.scala:297]: Can you add a test case where the list is empty and make sure the op is evaluated to `false`?

**Review by zhuansunxt — APPROVED** [review]
LGTM. 2 quick comments.

File: spark/src/main/scala/io/delta/sharing/spark/filters/OpConverter.scala (hunk 1 above)
Comment (chakankardb) [reply spark/src/main/scala/io/delta/sharing/spark/filters/OpConverter.scala:57]: Done.

File: spark/src/test/scala/io/delta/sharing/spark/filters/OpConverterSuite.scala (hunk 2 above)
Comment (chakankardb) [reply spark/src/test/scala/io/delta/sharing/spark/filters/OpConverterSuite.scala]: Spark will never send us an empty IN.
In fact, if you specify one entry in the IN operator in sql, spark converts that into an equality check in the expression tree it sends us.

However, I decided to handle both cases.
We throw an error if there are no predicates, and convert it to an equality if there is 1.

**Review by zhuansunxt — APPROVED** [review]
